packages/engine-server/src/markdown/remark/utils.ts

```diff
@@ -751,4 +785,116 @@ export class RemarkUtils {
       };
     };
   }
+
+  /** Extract all blocks from the note which could be referenced by a block anchor.
+   *
+   * If those blocks already have anchors (or if they are a header), this will also find that anchor.
+   *
+   * @param note The note from which blocks will be extracted.
+   */
+  static async extractBlocks({
+    note,
+    wsRoot,
+    engine,
+  }: {
+    note: NoteProps;
+    wsRoot: string;
+    engine: DEngineClient;
+  }): Promise<NoteBlock[]> {
+    const proc = MDUtilsV5.procRemarkFull({
+      engine,
+      vault: note.vault,
+      fname: note.fname,
+      dest: DendronASTDest.MD_DENDRON,
+    });
+    const slugger = getSlugger();
+
+    // Read and parse the note
+    // TODO: It might be better to get the text from the editor
```

Comment:
this is okay. we want to decouple, as much as makes sense, dendron fucntionality from vscode specific methods